eventstore/src/main/java/com/github/yafna/events/pipelines/AggregatePipeline.java

```diff
@@ -0,0 +1,113 @@
+package com.github.yafna.events.pipelines;
+
+import com.github.yafna.events.Event;
+import com.github.yafna.events.aggregate.Aggregate;
+import com.github.yafna.events.annotations.EvType;
+import com.github.yafna.events.annotations.Origin;
+import com.github.yafna.events.handlers.DomainHandlerRegistry;
+import com.github.yafna.events.store.EventStore;
+import com.google.gson.Gson;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Slf4j
+public class AggregatePipeline<A extends Aggregate> {
+    private final static String UNKNOWN_TYPE = "Unknown type";
+
+    private final Gson gson = new Gson();
+
+    private final EventStore store;
+    private final Map<String, A> objects = new HashMap<>();
+    private final String origin;
+    private final DomainHandlerRegistry<A> handlers;
+    private final Map<String, Class<?>> index;
+    private final Function<String, A> constructor;
+
+    /**
+     * @param clazz Aggregate class
+     * @param store event store to use
+     * @param eventTypes index map (event type name) -> (event class)
+     * @param handlers registry resolves event class to a list of handlers
+     * @param constructor function to create new aggregate instance for a given id
+     */
+    public AggregatePipeline(
+            Class<A> clazz,
+            EventStore store,
+            Map<String, Class<?>> eventTypes, DomainHandlerRegistry<A> handlers,
+            Function<String, A> constructor
+    ) {
+        origin = clazz.getAnnotation(Origin.class).value();
+        this.store = store;
+        this.handlers = handlers;
+        this.index = eventTypes;
+        this.constructor = constructor;
+    }
+
+
+    @SuppressWarnings("unused")
+    public <T> Event push(String aggregateId, T event) {
+        return store(event, aggregateId);
+    }
+
+    @SuppressWarnings("unused")
```

Comment:
This method contains more logic, rather than simple get. Could you please add brief comment ?